.github/workflows/build.yml

```diff
@@ -0,0 +1,48 @@
+name: Cache build artifacts
+
+on:
+  workflow_call:
```

Comment:
this is a reusable workflow, not a composite workflow

https://docs.github.com/en/actions/sharing-automations/reusing-workflows